rfcs/text/0000-rfc-resource-tagging.md

```diff
@@ -0,0 +1,200 @@
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# IaaS Resource Tagging for and Cost Monitoring and Auditing
+
+## Summary
+
+Define mandatory tagging and auditability requirements for DCP cloud (IaaS) resources and operator identities, to be used for:
+
+- Cost accounting, analysis, and monitoring;
+- Security auditing of administrative and operator actions.
+
+## Author(s)
+
+* [Amar Jandu](mailto:ajandu@ucsc.edu)
+* [Andrey Kislyuk](mailto:akisilyuk@chanzuckerberg.com)
+
+## Shepherd
+***Leave this blank.** This role is assigned by DCP PM to guide the **Author(s)** through the RFC process.*
+
+*Recommended format for Shepherds:*
+
+ `[Name](mailto:username@example.com)`
+
+## Motivation
+
+A substantial component of the HCA DCP budget is IaaS (cloud infrastructure) cost (specifically, AWS and GCP costs). We
+would like to monitor the sources of these costs on a service and component level. The separation of development and
+production IaaS accounts naturally allows us to track the cost breakdown on that dimension. AWS and GCP also provide
+a way to track costs by underlying IaaS service (such as S3 vs. EC2). However, we have no comprehensive way to separate
+costs by DCP service (such as Ingest Service vs. Data Store), development environment (dev vs. staging), or service component
+(such as DSS sync daemon vs. DSS checkout daemon).
+
+Monitoring costs by service and component will allow us to track down cost drivers among our components, prioritize
+engineering efforts to optimize our use of underlying cloud infrastructure, empower our developers to get daily feedback about
+the consequences of their serivce architecture design decisions, identify cost anomalies indicating possible abuse or security
+issues, and provide cost reporting, forecasting, and accountability to our funders.
+
+### User Stories
+
+* As a DCP developer, I would like to see how much a particular component of my service is costing in a particular test
+  environment, such as a routine scale test.
+* As a DCP operator, I would like to keep an eye on the running costs incurred by DCP services, to identify unexpected
+  cost increases or service availability issues.
+* As a DCP security lead, I would like to identify resource provisioning actions performed in the production account and
+  track them by operator identity and service.
+
+## Detailed Design
+
+We will use
+[AWS cost allocation tags](https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/custom-tags.html#allocation-how)
+and [GCP labels](https://cloud.google.com/resource-manager/docs/creating-managing-labels) to tag all resources that can
+be tagged. For resources that support tagging/labeling and constitute a significant portion of the IaaS budget (over 1%),
+tagging will be required. For other resources that support tagging/labeling, it is encouraged, but not required.
+
+#### AWS Services that Require Resource Tags
```

Comment:
I think it would be useful to include a section for GCP services that require tags, so that there is no confusion around which untagged assets the "reaper" process will be cleaning up.

Reply:
The DSS primarily uses cloudFunctions + buckets which fall into this category; are there other resources that you would recommend adding to this list?

Reply:
Thanks @amarjandu! I'd recommend adding GCE VMs since that is what we use to deploy the analysis service infrastructure and to run the pipelines.

Reply:
👍  I've added it to the list thanks!